### ISSUE TITLE
game service not getting ready

### ISSUE THREAD

**Issue by VikramTiwari** [tpryan/whack_a_pod#2]
Relevant logs:

Deployments:
```
kubectl get deployments
game-deployment    4         4         4            4           44m
```

Pods:
```
kubectl get pods
game-deployment-3498250528-5nzwt    1/1       Running   0          44m
game-deployment-3498250528-dxmlc    1/1       Running   0          44m
game-deployment-3498250528-g3rt0    1/1       Running   0          44m
game-deployment-3498250528-wxqd0    1/1       Running   0          44m
```

Services:
```
kubectl get services
game         10.27.253.177   <pending>        80:30826/TCP   42m
```

**Comment by tpryan** [comment]
what is the result of 'kubectl describe service game'

**Comment by VikramTiwari** [comment]
Ah! I see the error. Might have to rebuild from start?

```
$ kubectl describe service game
Name:			game
Namespace:		default
Labels:			name=game
Annotations:		kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"labels":{"name":"game"},"name":"game","namespace":"default"},"spec":{"loadBalancerIP"...
Selector:		app=game
Type:			LoadBalancer
IP:			10.27.253.177
Port:			http	80/TCP
NodePort:		http	30826/TCP
Endpoints:		10.24.0.15:8080,10.24.0.16:8080,10.24.1.11:8080 + 1 more...
Session Affinity:	None
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath	Type		Reason				Message
  ---------	--------	-----	----			-------------	--------	------				-------
  45m		3m		13	service-controller			Normal		CreatingLoadBalancer		Creating load balancer
  45m		3m		13	service-controller			Warning		CreatingLoadBalancerFailed	Error creating load balancer (will retry): Failed to create load balancer for service default/game: failed to create forwarding rule a9a5721d05dbc11e7870c42010a80007: googleapi: Error 400: Invalid value for field 'resource.IPAddress': '35.188.162.106'. Specified IP address is in-use and would result in a conflict., invalid
```

**Comment by tpryan** [comment]
Nope don't completely rebuild.  You just have to delete the forwarding rules.

**Comment by tpryan** [comment]
Easiest way to delete the forwarding rules.
Go to Cloud Console
Go to Networking. 
Go to External IPs
Find the IP you are managing, and there should be link in the "In Use by" to the fowarding rule that applies. 
Click the link. 
Delete.

**Comment by VikramTiwari** [comment]
Done. And now waiting for them to get back up. Will close the issue if all goes well.

**Comment by tpryan** [comment]
Cool, cool.

**Comment by VikramTiwari** [comment]
Services show that they are up. But nothing shows up while accessing the endpoint. Will debug later.

```
$ kubectl describe service api
Name:			api
Namespace:		default
Labels:			name=api
Annotations:		kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"labels":{"name":"api"},"name":"api","namespace":"default"},"spec":{"loadBalancerIP":"...
Selector:		app=api
Type:			LoadBalancer
IP:			10.27.241.90
LoadBalancer Ingress:	35.188.162.106
Port:			http	80/TCP
NodePort:		http	30854/TCP
Endpoints:		10.24.0.10:8080,10.24.0.11:8080,10.24.0.12:8080 + 9 more...
Session Affinity:	None
Events:			<none>

$ kubectl describe service admin
Name:			admin
Namespace:		default
Labels:			name=admin
Annotations:		kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"labels":{"name":"admin"},"name":"admin","namespace":"default"},"spec":{"loadBalancerI...
Selector:		app=admin
Type:			LoadBalancer
IP:			10.27.254.54
LoadBalancer Ingress:	35.188.132.74
Port:			http	80/TCP
NodePort:		http	32085/TCP
Endpoints:		10.24.1.13:8080
Session Affinity:	None
Events:			<none>

$ kubectl describe service game
Name:			game
Namespace:		default
Labels:			name=game
Annotations:		kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"labels":{"name":"game"},"name":"game","namespace":"default"},"spec":{"loadBalancerIP"...
Selector:		app=game
Type:			LoadBalancer
IP:			10.27.253.177
LoadBalancer Ingress:	35.188.162.106
Port:			http	80/TCP
NodePort:		http	30826/TCP
Endpoints:		10.24.0.15:8080,10.24.0.16:8080,10.24.1.11:8080 + 1 more...
Session Affinity:	None
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath	Type		Reason				Message
  ---------	--------	-----	----			-------------	--------	------				-------
  1h		51m		14	service-controller			Warning		CreatingLoadBalancerFailed	Error creating load balancer (will retry): Failed to create load balancer for service default/game: failed to create forwarding rule a9a5721d05dbc11e7870c42010a80007: googleapi: Error 400: Invalid value for field 'resource.IPAddress': '35.188.162.106'. Specified IP address is in-use and would result in a conflict., invalid
  1h		46m		15	service-controller			Normal		CreatingLoadBalancer		Creating load balancer
  46m		46m		1	service-controller			Normal		CreatedLoadBalancer		Created load balancer

```

**Comment by tpryan** [comment]
What does kubectl describe pod [name of a pod] say?

**Comment by VikramTiwari** [comment]
Pods are all up for each service. Here's what I have seen after some trial and error:

- When I start from scratch
  - 2 services get created `api` and `admin` and `game` gets conflicted
  - If I delete `api` and `admin` lbs then all three get created on k8s side but nothing on network tab in console
  - If I delete all the services, deployments, pods etc from the cluster, I can recreate this same behavior.

**Comment by tpryan** [comment]
Another typo:
GAMEIP=$(shell gcloud compute addresses list --filter='name=api AND region=$(REGION)'  --format='value[terminator=""](address)')
ADMINIP=$(shell gcloud compute addresses list --filter='name=admin AND region=$(REGION)'  --format='value[terminator=""](address)')
APIIP=$(shell gcloud compute addresses list --filter='name=api AND region=$(REGION)'  --format='value[terminator=""](address)') 

If you look, it's filtering 'name=api' to GAMEIP.

I fixed it in: #22e2c590e65c4d3ef2455ae773111de448d859fb

**Comment by VikramTiwari** [comment]
Sorry that I am just opening issues. Could have easily caught both of these and raised a PR. Thanks for following up on these though. :)
